### DISCUSSION TITLE
Cannot add button in updateOperation

### DISCUSSION THREAD

**Discussion by cod3rshotout** [Laravel-Backpack/community-forum#405] [Q&A (Help)] [answered]
Hello, I need some help.

I'm trying to add a "preview" button in the updateOperation, so I have the following content:

    public function setupUpdateOperation()
    {
        $this->crud->setUpdateContentClass('col-md-12');
        $this->setupCreateOperation('update');
        $this->crud->addButtonFromView('line', 'preview', 'preview', 'beginning');
    }

but the button doesn't appear at all, I have also created the blade file within this directory: `vendor/backpack/crud/buttons/preview.blade.php`, which have this content:

    <a href="/intranet/{{ $entry->slug }}?preview=1">
        <i class='fa fa-list'></i>Preview
    </a>

Essentially, I need to add a button near "Annulla":

<img src="https://i.imgur.com/kWDN8yy.jpg">

@pxpm @tabacitu any help?

**Comment by rchampagneca** [comment, marked as answer]
Modify a copy of **form_save_buttons.blade.php** from the vendor directory into **/resources/views/vendor/backpack/crud/inc**. Then include your custom button, as for example :

`@include('custom.buttons.mybutton')`

**Reply by tabacitu** [reply]
That should work great if you want to add the button for all CRUDs, yes. I'd only add one thing - that in order to still benefit from the changes Backpack might make inside `form_save_buttons.blade.php`, it's a good idea for you custom view to:
- include the view in Backpack's vendor dirctory
- then include your custom view

That way, you just _add something_, not really _override_ anything.

**Comment by tabacitu** [comment]
Hey @cod3rshotout - I realise it's unintuitive, but the `addButtonFromView()` method will only have effects on the List operation, not the Update operation. @rchampagneca 's solution above should work great!

**Reply by dsarmento** [reply]
Hi @tabacitu, it'd be great if we could add buttons to create/show/update views using addButton method.

**Reply by tabacitu** [reply]
I would freakin' love that too!

Maybe in our next version. Right now, the buttons that do exist in Create/Update are called SaveAction and behave differently. But even then... we would have to choose between:
- A) buttons alongside SaveActions;
- B) refactoring SaveActions so they are buttons;

I would not agree to B, not worth the breaking change. I would maaaybe agree to A... If the use cases were solid. And we'd find a way to document it so it's not confusing.

Which brings me to my question:

**What button did you need to add to the Create/Update page @dsarmento ? What did you want it to do?**

Cheers!

**Reply by dsarmento** [reply]
Hi @tabacitu, thanks for your answer.

I wanted to extended the PageManager add-on and add a preview page button/functionality. This would require me to add some sort of JS to get all the fields and POST them to a custom operation route.

I believe the SaveActions wouldn't allow me to do this because it saves the record first?

Do you have suggestions?

Thanks again.

**Reply by tabacitu** [reply]
Oh ok. Indeed with SaveActions you can only change what happens AFTER saving. so it's not a good fit.

But you can achieve that with a script widget: https://backpackforlaravel.com/docs/6.x/base-widgets 

Inside the script widget, you can add a little bit of JavaScript that adds a button wherever you want, and opens a page in a different tab for preview.
